pkg/v1/remote/transport/ping.go

```diff
@@ -63,31 +63,43 @@ func parseChallenge(suffix string) map[string]string {
 func ping(reg name.Registry, t http.RoundTripper) (*pingResp, error) {
 	client := http.Client{Transport: t}
 
-	url := fmt.Sprintf("%s://%s/v2/", reg.Scheme(), reg.Name())
-	resp, err := client.Get(url)
-	if err != nil {
-		return nil, err
+	// Always try https first, falling back to http if we can.
```

Comment:
I would do this here and propagate it to the other transports.  I don't think other transports should need to worry about fallback.

Reply:
Done, PTAL.